apps/server/src/controllers/userControllers/authUsers.ts

```diff
@@ -291,3 +299,69 @@ export const forgotPassword = async (req: Request, res: Response) => {
     })
   }
 }
+
+export const resetPassword = async (req: Request, res: Response) => {
+  try {
+    const { verify: longString } = req.query
+
+    // Check for reset-token instance using query-string(id)
+    const resetTokenInstance = await UserResetPasswordToken.findOne({
+      where: { id: longString as string, used: false },
+    })
+
+    if (!resetTokenInstance) {
+      return res.status(HTTP_STATUS_CODE.UNAUTHORIZED).json({
+        message: 'Invalid or already used verification token.',
+      })
+    }
+
+    const { id, token } = resetTokenInstance
+
+    // Verify the token with the jwt secret & query-string(id)
```

Comment:
you can replace this comment with 

    /**
     * References JWT secret in forgotPassword function
     * @see forgotPassword
     */
    const decodedToken = await Jwt.verify(token, {
      _secret: Env.JWT_SECRET + id,
    })